### PR TITLE
[memory][context] Block events mode

### PR DESCRIPTION
**Pull request by NikitaZotov**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/CONTRIBUTING.md)
* [x] Update changelog
* [ ] Update documentation

### CODE REVIEW

File: sc-memory/sc-core/sc_memory_context_manager.h [hunk 1]
```diff
@@ -122,4 +122,19 @@ void _sc_memory_context_pend_event(
  */
 void _sc_memory_context_emit_events(sc_memory_context const * ctx);
 
+//! Checks if specified sc-memory context has blocking events block.
+sc_bool _sc_memory_context_are_events_blocking(sc_memory_context const * ctx);
```
Comment (MksmOrlov) [inline sc-memory/sc-core/sc_memory_context_manager.h:126]: I think, we need this function in C++ API too

File: sc-memory/tests/sc-memory/events/test_sc_event.cpp [hunk 2]
```diff
@@ -408,3 +408,49 @@ TEST_F(ScEventTest, pend_events)
 
   EXPECT_EQ(passedCount, el_num);
 }
+
+TEST_F(ScEventTest, BlockEvents)
```
Comment (MksmOrlov) [inline sc-memory/tests/sc-memory/events/test_sc_event.cpp:412]: Add test where event is emitted. And where one event is emitted and after blocking not emitted

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

File: sc-memory/sc-core/sc_memory_context_manager.h (hunk 1 above)
Comment (NikitaZotov) [reply sc-memory/sc-core/sc_memory_context_manager.h:126]: It is private method. We can't use it in C API

File: sc-memory/sc-core/sc_memory_context_manager.c [hunk 3]
```diff
@@ -202,11 +203,11 @@ void _sc_memory_context_free_impl(sc_memory_context_manager * manager, sc_memory
   sc_monitor_release_write(&manager->context_monitor);
 }
 
-sc_bool _sc_memory_context_is_pending(sc_memory_context const * ctx)
+sc_bool _sc_memory_context_are_events_pending(sc_memory_context const * ctx)
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply sc-memory/sc-core/sc_memory_context_manager.c:206]: Events

[the remark this answers is not in the document]
Comment (MksmOrlov) [reply sc-memory/sc-core/sc_memory_context_manager.c:206]: sc-memory-context так написано

**Review by MksmOrlov — APPROVED** [review]

**Pull request merged**